modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/configuration/ClientExtensionConfiguration.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.client.extension.internal.configuration;
+
+import aQute.bnd.annotation.metatype.Meta;
+
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@ExtendedObjectClassDefinition(
+	category = "infrastructure",
+	scope = ExtendedObjectClassDefinition.Scope.COMPANY, strictScope = true
+)
+@Meta.OCD(
+	id = "com.liferay.client.extension.internal.configuration.ClientExtensionConfiguration",
+	localization = "content/Language",
+	name = "client-extension-configuration-name"
+)
+public interface ClientExtensionConfiguration {
+
+	@Meta.AD(deflt = "10000", name = "cluster-timeout", required = false)
+	public long clusterTimeout();
```

Comment:
I'm not sure this is the right naming here. I saw the other case we have for sync cluster invocation, and it's similar to this but via portal properties. So I guess we're fine, though a description would definitely help.

Reply:
Well, it will be inside "Client Extensions > Cluster Timeout", it's more or less evident. But yeah, I may add a description too saying that it's the timeout for deploy/undeploy cluster operations.